great_expectations/checkpoint/checkpoint.py

```diff
@@ -153,8 +153,9 @@ def run(
         if not self.validation_definitions:
             raise CheckpointRunWithoutValidationDefinitionError()
 
-        if not self.id:
-            self._add_to_store()
+        saved, errs = self.is_saved()
+        if not saved:
+            raise ValueError(errs)
```

Comment:
TODO: Use a custom type that aggregates errors and does a good job of displaying them in a readable manner